app/views/katello/subscription_mailer/subscription_expiry.html.erb

```diff
@@ -40,6 +40,9 @@
         </tr>
       <% end %>
     </table>
+  <% if @pools.any? %>
```

Comment:
Is this necessary? We don't send the email if there are no pools to begin with

Reply:
html.erb gets evaluated even if the pools are empty but the mail does not get delivered. I have moved the logic into the helper.